packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -100,17 +100,104 @@ abstract class SelectionHandler implements ValueListenable<SelectionGeometry> {
   SelectionResult dispatchSelectionEvent(SelectionEvent event);
 }
 
+/// A controller for the content of a [Selectable].
+///
+/// A user can set the value on this controller to modify the content under
+/// a selection created by [SelectionArea] or [SelectableRegion].
+abstract class SelectedContentController<T extends Object> {
```

Comment:
cc @LongCatIsLooong, this is the relevant code that we were discussing in the text input sync. A list of `SelectedContentControllers` is provided in `SelectionArea.onSelectionChanged` representing the active selection. The discussion was regarding whether this should be generic or specifically `TextSpan` since at the moment the frameworks only selectable widget is `Text`.